server/test/views/tags/IconElementTagProcessorTest.java

```diff
@@ -0,0 +1,88 @@
+package views.tags;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.matches;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import controllers.AssetsFinder;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.thymeleaf.context.ITemplateContext;
+import org.thymeleaf.model.IProcessableElementTag;
+import org.thymeleaf.processor.element.IElementTagStructureHandler;
+import play.Environment;
+import play.Mode;
+import repository.ResetPostgres;
+
+public class IconElementTagProcessorTest extends ResetPostgres {
+  @Mock private ITemplateContext mockContext;
+  @Mock private IProcessableElementTag mockTag;
+  @Mock private IElementTagStructureHandler mockStructureHandler;
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.openMocks(this);
+  }
+
+  @Test
+  public void renderTagWithClass() {
+    IconElementTagProcessor tagProcessor =
+        new IconElementTagProcessor(
+            "cf", instanceOf(AssetsFinder.class), instanceOf(Environment.class));
+    when(mockTag.getAttributeValue("type")).thenReturn("icon-https");
+    when(mockTag.hasAttribute("class")).thenReturn(true);
+    when(mockTag.getAttributeValue("class")).thenReturn("aaa bbb ccc");
+
+    tagProcessor.doProcess(mockContext, mockTag, mockStructureHandler);
+
+    String expected =
+        """
+<img src="/assets/Images/uswds/[a-zA-Z0-9]+-icon-https.svg" aria-hidden="true" role="img" class="aaa bbb ccc" alt="">
```

Comment:
Non-blocking Question: Why is [a-zA-Z0-9]+ part of the filepath?

Reply:
A fingerprint value is added to the path by the asset system. That regex accounts for it.